src/imgtools/cli/autopipeline.py

```
@@ -0,0 +1,245 @@
+    roi_map = {}
+    
+    # Load from YAML file if provided
+    if roi_match_yaml:
+        try:
+            with open(roi_match_yaml, 'r') as f:
+                yaml_data = yaml.safe_load(f)
+                if not isinstance(yaml_data, dict):
+                    raise click.BadParameter(f"ROI match YAML must contain a dictionary")
+                roi_map = yaml_data
+        except Exception as e:
+            raise click.BadParameter(f"Error reading ROI match YAML file: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)